postgres.go

```diff
@@ -0,0 +1,374 @@
+package eventmaster
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/lib/pq"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	eventmaster "github.com/wish/eventmaster/proto"
+)
+
+// implements datastore
+type PostgresStore struct {
+	db *sql.DB
+}
+
+type PostgresConfig struct {
+	Addr        string `json:"addr"`
+	Port        int    `json:"port"`
+	Database    string `json:"database"`
+	ServiceName string `json:"service_name"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+}
+
+func NewPostgresStore(c PostgresConfig) (*PostgresStore, error) {
+	var host string
+	if c.ServiceName != "" {
+		host = c.ServiceName + ".service.consul"
+	} else {
+		host = c.Addr
+	}
+	log.Infof("Connecting to postgres: %v", host)
+	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
+		"password=%s dbname=%s sslmode=disable",
+		host, c.Port, c.Username, c.Password, c.Database)
+	db, err := sql.Open("postgres", psqlInfo)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating postgres session")
+	}
+	err = db.Ping()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating postgres session")
+	}
+	log.Infof("Successfully connected to postgres %s", host)
+
+	return &PostgresStore{
+		db: db,
+	}, nil
+}
+
+func (p *PostgresStore) AddEvent(e *Event) error {
```

Comment:
let's add some comments for all the public functions

Reply:
Done